Reduce arguments

howardbaik opened the pull request:
<!--This PR Template was modified from https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/.github/PULL_REQUEST_TEMPLATE.md-->

#### Purpose/implementation Section

Reduce the number of arguments to `ari_narrate()` and `ari_stitch()` by consolidating all the ffmpeg arguments into one list, called `ffmpeg_args`. 

#### What changes are being implemented in this Pull Request?

Create new function argument, `ffmpeg_args`, which is a list of all the ffmpeg options. Then, refer to these options by accessing the list elements, for ex, `ffmpeg_args$video_filters`.

Commit:
Put all the ffmpeg related arguments into a list called `ffmpeg_args`

Commit:
Syntax fix

howardbaik committed:
Merge branch 'ariExtra-immigration' of https://github.com/jhudsl/ari into reduce-arguments

howardbaik committed:
Run `document()`

seankross commented on R/ari_stitch.R:
```diff
@@ -74,44 +74,40 @@ ari_stitch <- function(images, audio,
+                                          video_sync_method = "2",
+                                          pixel_format = "yuv420p",
+                                          fast_start = FALSE,
+                                          stereo_audio = TRUE),
```
Need to create a new function called set_ffmpeg_args():

```r
set_ffmpeg_args <- function(){
 list(frames_per_second = NULL,
  video_filters = NULL,
  divisible_height = TRUE,
  audio_codec = get_audio_codec(),
  video_codec = get_video_codec(),
  deinterlace = FALSE,
  ffmpeg_opts = "",
  audio_bitrate = NULL,
  video_bitrate = NULL,
  video_sync_method = "2",
  pixel_format = "yuv420p",
  fast_start = FALSE,
  stereo_audio = TRUE)
}
```

Question for you is do you want this function to take arguments? (Could be something you don't get around to until the future.)

seankross commented on R/ari_stitch.R:
```diff
@@ -194,68 +188,69 @@ ari_stitch <- function(images, audio,
-  ffmpeg_opts <- c(video_filters, ffmpeg_opts)
+  ffmpeg_opts <- c(video_filters, ffmpeg_args$ffmpeg_opts)
   ffmpeg_opts <- paste(ffmpeg_opts, collapse = " ")
 
```
Applies starting at 195: any ffmpeg argument logic should be handled by set_ffmpeg_args(), ideally: create a separate function that takes the result of executing set_ffmpeg_args() and constructs the result into a system command. (non-exported)

seankross commented on R/ari_stitch.R:
```diff
@@ -194,68 +188,69 @@ ari_stitch <- function(images, audio,
+    ifelse(!is.null(ffmpeg_args$frames_per_second), paste0("-r ", ffmpeg_args$frames_per_second), ""),
     ifelse(experimental, "-strict experimental", ""),
     "-max_muxing_queue_size 9999",
     "-threads 2",
```
```
system(build_ffmpeg_command(ffmpeg_args))
```

seankross commented on R/ari_stitch.R:
```diff
@@ -74,44 +74,40 @@ ari_stitch <- function(images, audio,
                        output = tempfile(fileext = ".mp4"),
```
`output` should have no default value

howardbaik committed:
Create `set_ffmpeg_args()`

howardbaik committed:
Build ffmpeg command to supply to `system()`

howardbaik committed:
Resolve merge conflicts with `ariExtra-immigration` branch

Merge branch 'ariExtra-immigration' into reduce-arguments

# Conflicts:
#	R/ari_narrate.R
#	R/ari_spin.R
#	man/ari_narrate.Rd

howardbaik committed:
More Documentation stuff
